### PR TITLE
CreateCommonKrylovSolverSetup can specify `abs_tol` and `rel_tol`

### LINKED ISSUE #168: `CreateCommonKrylovSolverSetup` lies about the tolerance parameter
The API has this signature:

https://github.com/Open-Sn/opensn/blob/2bfe5cfd281a33089abff793218c4fa68b1fdcf4/framework/math/petsc_utils/petsc_utils.cc#L96-L103
And then `in_relative_residual_tolerance` argument is used like this:
https://github.com/Open-Sn/opensn/blob/2bfe5cfd281a33089abff793218c4fa68b1fdcf4/framework/math/petsc_utils/petsc_utils.cc#L115-L116

The `KSPSetTolerances` signature is:
```
PetscErrorCode KSPSetTolerances(KSP ksp, PetscReal rtol, PetscReal abstol, PetscReal dtol, PetscInt maxits)
```
See [doco](https://petsc.org/release/manualpages/KSP/KSPSetTolerances/)

Notice that `in_relative_residual_tolerance` is being passed into PETSc as **absolute** one 🤦‍♂️

### PR DESCRIPTION
**Pull request by andrsd**
Closes #168

### CODE REVIEW

**Review by wdhawkins — APPROVED** [review]

File: framework/math/petsc_utils/petsc_utils.cc [hunk 1]
```diff
@@ -98,7 +98,8 @@ CreateCommonKrylovSolverSetup(Mat ref_matrix,
                               const std::string& in_solver_name,
                               const std::string& in_solver_type,
                               const std::string& in_preconditioner_type,
-                              double in_relative_residual_tolerance,
+                              double in_rel_tol,
+                              double in_abs_tol,
```
Comment (zhardy-lanl) [inline framework/math/petsc_utils/petsc_utils.cc:102]: Just making note of this for future reference: We should get rid of the `in_<var>` argument names.

File: modules/cfem_diffusion/cfem_diffusion_solver.cc [hunk 2]
```diff
@@ -336,6 +336,7 @@ Solver::Execute()
                                   TextName(),
                                   KSPCG,
                                   PCGAMG,
+                                  0.,
```
Comment (zhardy-lanl) [inline modules/cfem_diffusion/cfem_diffusion_solver.cc:339]: Is there a preference in the codebase for `0.` vs `0.0` in our standards? I believe the codebase primarily uses the trailing zero (which is also my preference). Not requesting a change here, but `0.0` would be welcomed where `0.` was introduced in various spots this PR.

**Review by zhardy-lanl — APPROVED** [review]
Please see my comment on `0.0` vs. `0.` and change if you would feel necessary. Good to go other than that.

File: modules/cfem_diffusion/cfem_diffusion_solver.cc (hunk 2 above)
Comment (andrsd) [reply modules/cfem_diffusion/cfem_diffusion_solver.cc:339]: > Is there a preference in the codebase for 0. vs 0.0 in our standards?

I do not think there is anything in the code standards regarding this.

Comment (andrsd) [reply modules/cfem_diffusion/cfem_diffusion_solver.cc:339]: There is a small amount of `0.` or `<number>.` in the code. I will update this PR and send another one to unify this. Plus a change to code standards.

File: framework/math/petsc_utils/petsc_utils.cc (hunk 1 above)
Comment (andrsd) [reply framework/math/petsc_utils/petsc_utils.cc:102]: Rough search shows about 300 occurrences. Opened issue #173 for this task.

**Pull request merged**